src/py_zfs_pool.c

```diff
@@ -28,378 +24,257 @@ int py_zfs_pool_init(PyObject *type, PyObject *args, PyObject *kwds) {
 
 static
 void py_zfs_pool_dealloc(py_zfs_pool_t *self) {
+	if (self->zhp != NULL) {
+		Py_BEGIN_ALLOW_THREADS
+		zpool_close(self->zhp);
+		Py_END_ALLOW_THREADS
+		self->zhp = NULL;
+	}
+	Py_CLEAR(self->name);
+	Py_CLEAR(self->pylibzfsp);
 	Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
 static
-PyObject *py_zfs_pool_asdict(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_reiterate_vdevs(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_unsup_features(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_wait(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
+PyObject *py_zfs_pool_get_name(py_zfs_pool_t *self, void *extra) {
+	if (self == NULL)
+		Py_RETURN_NONE;
+	return Py_NewRef(self->name);
 }
 
 static
-PyObject *py_zfs_pool_sync(PyObject *self, PyObject *args) {
+PyObject *py_zfs_pool_asdict(PyObject *self, PyObject *args) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_create(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_root_dataset(PyObject *self, PyObject *args) {
+	py_zfs_dataset_t *out = NULL;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	PyObject *fargs = PyTuple_New(0);
+	PyObject *fkwargs = PyDict_New();
```

Comment:
Maybe:

```
Py_BuildValue("{s:s}", "name", p->name);
```

Reply:
If `p->name` is a unicode object
```
Py_BuildValue("{s:O}", "name", p->name);
```
Which will automatically incref the object.

Reply:
Thanks, I have updated.